### PR TITLE
Edit scheduled newcomers

### LINKED ISSUE #1009: Edit scheduled newcomers
1) Not to forget: re-schedule task to create user account
2) Delete task if entry is deleted
3) Delete user account request

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #1009

## PR includes

-   [x] Feature

### CODE REVIEW

File: src/trpc/router/userCreationRequestRouter.ts [hunk 1]
```diff
@@ -47,6 +48,41 @@ export const userCreationRequestRouter = router({
...
+            ctx.session.user.id,
+        );
+        const servicesBefore = userCreationRequestBefore.services as { serviceId: string; serviceName: string }[];
+
+        const phoneBefore = servicesBefore.find((service) => service.serviceName === 'Phone')?.serviceId;
+
+        await historyEventMethods.create({ user: ctx.session.user.id }, 'editUserCreationRequest', {
+            groupId: undefined,
+            userId: ctx.session.user.id,
+            before: {
```
Comment (tsumo) [inline src/trpc/router/userCreationRequestRouter.ts:67]: Use `dropUnchanged` here

File: src/trpc/router/userCreationRequestRouter.ts [hunk 2]
```diff
@@ -47,6 +48,41 @@ export const userCreationRequestRouter = router({
         return creationRequest;
     }),
 
+    edit: protectedProcedure.input(editUserCreationRequestSchema).mutation(async ({ input, ctx }) => {
+        accessCheck(checkRoleForAccess(ctx.session.user.role, 'editUserCreationRequests'));
```
Comment (tsumo) [inline src/trpc/router/userCreationRequestRouter.ts:52]: Should we check if request is denied here?

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/EditUserCreationRequestModal.tsx/EditUserCreationRequestModal.tsx.i18n/ru.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "Edit profile creation request for {nameAndOrganization}": "Редактировать заявку на создание профиля для {nameAndOrganization}",
+    "Phone": "Телефон",
+    "Date": "Дата выхода",
+    "Email": "",
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/EditUserCreationRequestModal.tsx/EditUserCreationRequestModal.tsx.i18n/ru.json:5]: Why? we have form to change email here

File: src/components/HistoryRecord/HistoryRecord.tsx [hunk 4]
```diff
@@ -37,7 +37,7 @@ const ChangeListItem = ({
     before?: string | number | boolean;
     after?: string | number | boolean;
 }) => {
-    if (after === undefined) return null;
+    if (after === undefined || after === before) return null;
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/HistoryRecord/HistoryRecord.tsx:40]: thanks, forgot about it

File: src/modules/historyEventTypes.ts [hunk 5]
```diff
@@ -84,6 +84,18 @@ interface HistoryEventsData {
             comment?: string;
         };
     };
+    editUserCreationRequest: {
+        requireGroupId: false;
+        requireUserId: true;
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/modules/historyEventTypes.ts:89]: thought it about author of event:/

File: src/trpc/router/userCreationRequestRouter.ts (hunk 2 above)
Comment (nyet-ty) [reply src/trpc/router/userCreationRequestRouter.ts:52]: good idea, thanks

Comment (nyet-ty) [reply src/trpc/router/userCreationRequestRouter.ts:55]: added check of request state below

File: src/components/EditUserCreationRequestModal.tsx/EditUserCreationRequestModal.tsx.i18n/ru.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (tsumo) [reply src/components/EditUserCreationRequestModal.tsx/EditUserCreationRequestModal.tsx.i18n/ru.json:5]: I mean it doesn't need a translation

File: src/modules/userCreationRequestMethods.ts [hunk 6]
```diff
@@ -204,6 +203,90 @@ export const userCreationRequestsMethods = {
...
+        });
+
+        if (restData.date && requestBeforeUpdate.date !== restData.date) {
+            const { users, to: mailTo } = await userMethods.getMailingList(
+                'createScheduledUserRequest',
+                updatedRequest.organizationUnitId,
+                updatedRequest.creator ? updatedRequest.creator : undefined,
+            );
+
+            const icalSublect = `${updatedRequest.creationCause === 'transfer' ? tr('Transfer') : tr('Employment')} ${
```
Comment (tsumo) [inline src/modules/userCreationRequestMethods.ts:258]: Subject

File: src/trpc/router/userCreationRequestRouter.ts [hunk 7]
```diff
@@ -47,6 +51,48 @@ export const userCreationRequestRouter = router({
...
+        const userCreationRequestBefore = await userCreationRequestsMethods.getById(input.id);
+
+        if (userCreationRequestBefore.status === 'Denied') {
```
Comment (tsumo) [inline src/trpc/router/userCreationRequestRouter.ts:58]: Good, but let's move it from router into a method

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/EditUserCreationRequestModal.tsx/EditUserCreationRequestModal.tsx.i18n/ru.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/EditUserCreationRequestModal.tsx/EditUserCreationRequestModal.tsx.i18n/ru.json:5]: Just use text in the form, not translation, get it

**Review by tsumo — APPROVED** [review]

**Comment by IgorGoryany** [comment]
Build failed

**Comment by IgorGoryany** [comment]
Build failed

**Review by asabotovich — APPROVED** [review]

**Comment by asabotovich** [comment]
Build failed

**Pull request merged**